core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,8 +389,18 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            if (metadata.getView(session, tableName).isEmpty() &&
-                    metadata.getTableHandle(session, tableName).isEmpty()) {
+
+            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+            if (tableHandle.isEmpty()) {
+                throw semanticException(TABLE_NOT_FOUND, showColumns, "Table '%s' does not exist", tableName);
+            }
+            else {
+                // Queries against information_schema should never fail, so in case of DESC is there an issue in getting
+                // table metadata the error could be swallowed. To avoid this we call here getTableMetadata.
+                metadata.getTableMetadata(session, tableHandle.orElseThrow());
```

Comment:
Now that we got table metadata,  we could provide `VALUES` out of it.
No need to delegate back to information_schema to produce something we already know

Reply:
Then I would need to handle predicate of `SHOW COLUMNS LIKE '%some_complex%regexp'`

Reply:
like stashing a Filter on top of Values? i guess we do this anyway

Reply:
It got quite complex...